OSDOCS-20005:Update the z-stream RNs for 4.21.19 (#112842)
openshift/openshift-docs: modules/zstream-4-21-19.adoc

:_mod-docs-content-type: REFERENCE
[id="zstream-4-21-19_{context}"]
= RHSA-2026:23241 - {product-title} {product-version}.19 fixed issues advisory

Issued: 11 June 2026

[role="_abstract"]
{product-title} release {product-version}.19 is now available. The list of fixed issues that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2026:23241[RHSA-2026:23241] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2026:23239[RHBA-2026:23239] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.21.19 --pullspecs
----

[id="zstream-4-21-19-fixed-issues_{context}"]
== Fixed issues

* Before this update, pagination controls were not present at mobile resolutions because PatternFly expected both top and bottom pagination controls to be used. With this release, pagination controls are present regardless of the mobile resolution. (link:https://redhat.atlassian.net/browse/OCPBUGS-84967[OCPBUGS-84967])

> * Before this update, the manila container storage interface (CSI) driver node plugin might have crashed on startup if the Network File System (NFS) CSI plugin socket was not yet available, This issue might have occurred, for example, after a node reboot. With this release, the manila CSI node DaemonSet waits for the NFS plugin socket to be ready before starting the driver, which prevents crash loops due to transient startup ordering. (link:https://redhat.atlassian.net/browse/OCPBUGS-85572[OCPBUGS-85572])

Reviewer: ```suggestion
* Before this update, the manila container storage interface (CSI) driver node plugin sometimes crashed on startup if the Network File System (NFS) CSI plugin socket was not yet available. For example, this issue sometimes occurred after a node reboot. With this release, the manila CSI node DaemonSet waits for the NFS plugin socket to be ready before starting the driver, which prevents crash loops due to transient startup ordering. (link:https://redhat.atlassian.net/browse/OCPBUGS-85572[OCPBUGS-85572])
```

> * Before this update, a change in the `controller-runtime` library in {product-title} 4.21 caused logging not to be initialized for some of the controllers for the Operator. As a consequence, the Cluster Ingress Operator did not log controller initialization or reconciliation errors for the `gateway-labeler`,  `gateway-service-dns`, and `gatewayclass` controllers. With this release, the Operator is updated to initialize logging for these controllers. As a result, the Operator logs initialization and reconciliation errors for these controllers. (link:https://issues.redhat.com/browse/OCPBUGS-86027[OCPBUGS-86027])

Reviewer: ```suggestion
* Before this update, a change in the `controller-runtime` library in {product-title} 4.21 caused logging to not be initialized for some of the controllers for the Operator. As a consequence, the Cluster Ingress Operator did not log controller initialization or reconciliation errors for the `gateway-labeler`,  `gateway-service-dns`, and `gatewayclass` controllers. With this release, the Operator is updated to initialize logging for these controllers. As a result, the Operator logs initialization and reconciliation errors for these controllers. (link:https://issues.redhat.com/browse/OCPBUGS-86027[OCPBUGS-86027])
```

* Before this update, users without any projects received a `Restricted access` error message when they navigated to certain resource list pages (for example, the *Pods*, *PodDisruptionBudgets*, *RoleBindings*, *VolumeSnapshots*, and *Helm* pages). With this release, these pages correctly display an empty state instead of a misleading `403` error. (link:https://issues.redhat.com/browse/OCPBUGS-86314[OCPBUGS-86314])

* Before this update, when you navigated to the resource detail pages for Shipwright components (`Build`, `BuildRun`, `BuildStrategy`, or `ClusterBuildStrategy`) with the builds for the {product-title} Operator installed, the console crashed with React error `#310`. With this release, the component rendering logic is corrected. As a result, Shipwright detail pages load successfully without crashing. (link:https://issues.redhat.com/browse/OCPBUGS-86427[OCPBUGS-86427])

> * Before this update, the Hosted Cluster Config Operator (HCCO) globalps and resources controllers did not have watches on the control-plane namespace pull secret. When you updated the Secret data referenced by the `HostedCluster.spec.pullSecret` parameter in-place (without changing the Secret name), the controllers had no way to immediately detect the change. As a consequence, on {aws-first} and {azure-first} clusters that used Replace upgrade type `NodePools`, in-place pull secret updates were not promptly propagated to the guest cluster. The `kube-system/original-pull-secret` and `openshift-config/pull-secret` secrets in the guest cluster reconciled only on the next periodic sync cycle, which caused a significant delay before updated registry credentials became available to the hosted cluster workloads. With this release, control-plane namespace pull-secret watches are added to the globalps and resources HCCO controllers so that they are notified immediately when the pull secret data changes in the control plane namespace. As a result, in-place updates to the pull secret referenced by the `HostedCluster.spec.pullSecret` parameter promptly propagate to `openshift-config/pull-secret`, `kube-system/original-pull-secret`, and the kubelet `config.json` parameter in the guest cluster through the Global Pull Secret DaemonSet, without requiring a `NodePool` rollout or the need to wait for a periodic resync. (link:https://issues.redhat.com/browse/OCPBUGS-86477[OCPBUGS-86477])

Reviewer: ```suggestion
* Before this update, the `globalps` and `resources` controllers for the Hosted Cluster Config Operator did not have watches on the control-plane namespace pull secret. For example, when you updated the Secret data referenced by the `HostedCluster.spec.pullSecret` parameter in-place without changing the Secret name, the controllers did not immediately detect the change. As a consequence, on {aws-first} and {azure-first} clusters that used `NodePool` objects with the Replace upgrade type, in-place pull secret updates were not promptly propagated to the hosted cluster. The `kube-system/original-pull-secret` and `openshift-config/pull-secret` secrets in the hosted cluster were reconciled only on the next periodic sync cycle, which caused a significant delay before updated registry credentials became available to the hosted cluster workloads. With this release, control-plane namespace pull-secret watches are added to the `globalps` and `resources` controllers so that they are notified immediately when the pull secret data changes in the control plane namespace. As a result, in-place updates to the pull secret referenced by the `HostedCluster.spec.pullSecret` parameter are propagated to `openshift-config/pull-secret`, `kube-system/original-pull-secret`, and the kubelet `config.json` parameter in the hosted cluster through the Global Pull Secret DaemonSet without needing a `NodePool` rollout or waiting for a periodic resync. (link:https://issues.redhat.com/browse/OCPBUGS-86477[OCPBUGS-86477])
```
I provided a suggested edit to clear up some of the hosted control planes terminology. I was curious about this one, because it sounds like an internal bug. The OCPBUG is marked as "Release Note Not Required", so you might want to double-check that this is a customer-facing issue.

Author: I missed that it was `Release Note Not Required`. I removed this text from the PR.

> * Before this update, the `python3-perf` RPM package was incorrectly pinned to an old version. As a result, the RPM package was being flagged by security scanners for including more than 400 outstanding CVEs because the `python-3-perf` RPM package was built from the kernel RPM package. With this release, the pinning issue is resolved. As a result, the scanning result will be clean. (link:https://issues.redhat.com/browse/OCPBUGS-86513[OCPBUGS-86513])

Reviewer: ```suggestion
* Before this update, the `python3-perf` RPM package was incorrectly pinned to an old version. As a result, the RPM package was being flagged by security scanners for including more than 400 outstanding CVEs because the `python-3-perf` RPM package was built from the kernel RPM package. With this release, the pinning issue is resolved. As a result, the scanning result is clean. (link:https://issues.redhat.com/browse/OCPBUGS-86513[OCPBUGS-86513])
```

> * Before this update, a failure to update the `PerformanceProfile` status due to a temporarily unavailable API server during an upgrade or under network load could leave the profile in a `Degraded` condition. As a consequence, the degraded condition might get stuck and never resolve, even after the cluster returned to a healthy state, because the Operator did not retry until the next reconcile event. With this release, the Operator schedules a retry of the status update until it succeeds, retrying approximately every 30 seconds. As a result, the stuck degraded status is temporary and resolves automatically during the next retry. (link:https://issues.redhat.com/browse/OCPBUGS-86809[OCPBUGS-86809])

Reviewer: ```suggestion
* Before this update, if an API server was temporarily unavailable because of an upgrade or network load, the `PerformanceProfile` status was not updated. As a consequence, the degraded condition might get stuck and never resolve, even after the cluster returned to a healthy state, because the Operator did not retry until the next reconcile event. With this release, the Operator schedules a retry of the status update until it succeeds, retrying approximately every 30 seconds. As a result, the stuck degraded status is temporary and is resolved automatically during the next retry. (link:https://issues.redhat.com/browse/OCPBUGS-86809[OCPBUGS-86809])
```
